tst/EnergyPlus/unit/ElectricPowerServiceManager.unit.cc

```diff
@@ -1230,3 +1230,148 @@ TEST_F(EnergyPlusFixture, Battery_checkUserEfficiencyInputTest)
+        "  -100,                    !- Minimum Curve Output",
+        "  100,                     !- Maximum Curve Output",
+        "  Dimensionless,           !- Input Unit Type for x",
+        "  Dimensionless;           !- Output Unit Type",
+
+        "Curve:RectangularHyperbola2,",
+        "  discharging1,            !- Name",
+        "  0.25,                    !- Coefficient1 C1",
+        "  -4.0,                    !- Coefficient2 C2",
+        "  -2.0,                    !- Coefficient3 C3",
+        "  0,                       !- Minimum Value of x",
+        "  1,                       !- Maximum Value of x",
+        "  -100,                    !- Minimum Curve Output",
+        "  100,                     !- Maximum Curve Output",
+        "  Dimensionless,           !- Input Unit Type for x",
+        "  Dimensionless;           !- Output Unit Type",
+
+        "Curve:RectangularHyperbola2,",
+        "  charging2,               !- Name",
+        "  -.2765,                  !- Coefficient1 C1",
+        "  -93.27,                  !- Coefficient2 C2",
+        "  0.0068,                  !- Coefficient3 C3",
+        "  0,                       !- Minimum Value of x",
+        "  1,                       !- Maximum Value of x",
+        "  -100,                    !- Minimum Curve Output",
+        "  100,                     !- Maximum Curve Output",
+        "  Dimensionless,           !- Input Unit Type for x",
+        "  Dimensionless;           !- Output Unit Type",
+
+        "Curve:RectangularHyperbola2,",
+        "  discharging2,            !- Name",
+        "  0.0899,                  !- Coefficient1 C1",
+        "  -98.24,                  !- Coefficient2 C2",
+        "  -.0082,                  !- Coefficient3 C3",
+        "  0,                       !- Minimum Value of x",
+        "  1,                       !- Maximum Value of x",
+        "  -100,                    !- Minimum Curve Output",
+        "  100,                     !- Maximum Curve Output",
+        "  Dimensionless,           !- Input Unit Type for x",
+        "  Dimensionless;           !- Output Unit Type",
+
+    });
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    Curve::GetCurveInput(*state);
+    state->dataCurveManager->GetCurvesInputFlag = false;
+
+    // Test 1: Curves which do not produce errors
+    e0c = 13.0;
+    e0d = 11.0;
+    chaCurveIndex = 1;
+    disCurveIndex = 2;
+
+    checkChargeDischargeVoltageCurves(*state, "Test 1", e0c, e0d, chaCurveIndex, disCurveIndex);
+
+    EXPECT_TRUE(compare_err_stream(""));
+
+    // Test 2: Curves from ShopwithPVBattery.idf as of 7/2023
+    e0c = 12.6;
+    e0d = 12.4;
+    chaCurveIndex = 3;
+    disCurveIndex = 4;
+
+    checkChargeDischargeVoltageCurves(*state, "Test 2", e0c, e0d, chaCurveIndex, disCurveIndex);
+
+    std::string const error_string2 = delimited_string(
+        {"   ** Warning ** Kinetic Battery Model: Test 2 has a charging/discharging voltage curve conflict.",
```

Comment:
You should probably add the battery name in case there are multiple batteries.

Reply:
Oh, `Test 2` is the battery name.

Reply:
Yup--Test 2 is the battery name.  The only reason it is passed is for better reporting if there is a problem.

Reply:
I've changed the name to "Battery for Test 2" so it's clearer hopefully.